kratos/python_scripts/apply_periodic_boundary_condition_process.py

```diff
@@ -0,0 +1,120 @@
+
+    This class constructs the model parts containing the constraints that enforce the rigid body movement
+    Only the member variables listed below should be accessed directly.
+
+    Public member variables:
+    Model -- the container of the different model parts.
+    settings -- Kratos parameters containing solver settings.
+    """
+
+    def __init__(self, Model, settings ):
+        """ The default constructor of the class
+
+        Keyword arguments:
+        Model -- the container of the different model parts.
+        settings -- Kratos parameters containing solver settings.
+        """
+        KratosMultiphysics.Process.__init__(self)
+
+        default_settings = KratosMultiphysics.Parameters("""
+        {
+            "help"                        : "This process uses LinearMasterSlaveConstraint in order to impose a periodic boundary condition on the given submodelparts. The process takes the first provided submodelpart as master and the second as slave.",
+            "computing_model_part_name"   : "",
+            "model_part_name"             : "please_specify_model_part_name",
+            "first_model_part_name"       : "please_specify_model_part_name",
+            "second_model_part_name"      : "please_specify_model_part_name",
+            "interval"                    : [0.0, 1e30],
+            "transformation_settings"     : {
+                                                "rotation_settings":{
+                                                    "center":[0,0,0],
+                                                    "axis_of_rotation":[0.0,0.0,0.0],
+                                                    "angle_degree":0.0
+                                                },
+                                                "translation_settings":{
+                                                    "dir_of_translation":[0.0,0.0,0.0],
+                                                    "magnitude":0.0
+                                                }
+                                            },
+            "search_settings"             : {
+                                                "max_results":100000,
+                                                "tolerance": 1E-6
+                                            },
+            "variable_names"              : []
+        }
+        """)
+
+        # Detect "End" as a tag and replace it by a large number
+        if(settings.Has("interval")):
+            if(settings["interval"][1].IsString()):
+                if(settings["interval"][1].GetString() == "End"):
+                    settings["interval"][1].SetDouble(1e30) # = default_settings["interval"][1]
+                else:
+                    raise Exception("The second value of interval can be \"End\" or a number, interval currently:"+settings["interval"].PrettyPrintJsonString())
+
+        settings.ValidateAndAssignDefaults(default_settings)
+
+        if(settings.Has("variable_names")):
```

Comment:
this check is not necessary after the validation

Reply:
not necessary .. removing !